include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4630,6 +4630,21 @@ struct OrtApi {
    * \snippet{doc} snippets.dox OrtStatus Return Value
    */
   ORT_API2_STATUS(KernelInfoGetAllocator, _In_ const OrtKernelInfo* info, _In_ OrtMemType mem_type, _Outptr_ OrtAllocator** out);
+
+  /** \brief Set external initializer file buffer
+   *
+   * \param[in] session options
+   * \param[in] external_file_names Array of null terminated UTF-8 encoded strings of the external initializer file names.
+   * \param[in] array_buffer Array of pointers to the buffer of the file context
+   * \param[in] file_lengths Array of size_t to indicate the length of file context
+   * \param[in] initializers_num Number of elements in the initializer_names and initializers
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   */
+  ORT_API2_STATUS(AddExternalInitializerFiles, _In_ OrtSessionOptions* options,
+                  _In_reads_(input_len) const ORTCHAR_T* const* external_file_names,
+                  _In_reads_(input_len) void* const* array_buffer,
+                  _In_reads_(input_len) const size_t* file_lengths, size_t initializers_num);
```

Comment:
should the `input_len` here be "file_lengths"?

Reply:
what is the expected life-cycle for the array buffers? When can users expect to safely free the memories? Those information may be important and worth to be documented.

Reply:
should be initializers_num, will update it.

Reply:
yes, will do, user can free the memory after session created.